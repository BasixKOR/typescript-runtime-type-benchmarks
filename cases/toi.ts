import * as toi from '@toi/toi'
import { Data } from '../data'

const isValid = toi
  .required()
  .and(toi.obj.isplain())
  .and(
    toi.obj.keys({
      number: toi.num.is(),
      neg_number: toi.num.max(0),
      max_number: toi.num.max(Number.MAX_VALUE),
      string: toi.str.is(),
      long_string: toi.str.is().and(toi.str.min(100)),
      boolean: toi.bool.is(),
      deeplyNested: toi
        .required()
        .and(toi.obj.isplain())
        .and(toi.obj.keys({
          foo: toi.str.is(),
          num: toi.num.is(),
          bool: toi.bool.is()
        }))
    })
  )

export function caseToi(data: Data) {
  return isValid(data)
}