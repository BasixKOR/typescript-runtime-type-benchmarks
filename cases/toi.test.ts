import { caseToi } from './toi'
import { DATA } from '../data'

describe('toi', () => {
  it('should pass', () => {
    const res = caseToi(DATA)
    expect(res).toEqual(DATA)

    if (res) {
      expect(res.number).toEqual(DATA.number)
    }
  })

  it('should fail negative number test', () => {
    const req = Object.assign({}, DATA, { neg_number: 1 })
    expect(() => caseToi(req)).toThrowErrorMatchingSnapshot()
  })

  it('should fail min length test', () => {
    const req = Object.assign({}, DATA, { long_string: '' })
    expect(() => caseToi(req)).toThrowErrorMatchingSnapshot()
  })
})
