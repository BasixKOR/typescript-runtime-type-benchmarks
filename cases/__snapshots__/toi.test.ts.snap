// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toi should fail min length test 1`] = `"value does not match structure"`;

exports[`toi should fail negative number test 1`] = `"value does not match structure"`;
